client/ayon_shotgrid/plugins/publish/collect_shotgrid_session.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import ayon_api
+from ayon_shotgrid.lib.credentials import get_local_login
+
+import pyblish.api
+import shotgun_api3
+from shotgun_api3.shotgun import AuthenticationFault
+
+
+class CollectShotgridSession(pyblish.api.ContextPlugin):
+    """Collect shotgrid session using user credentials"""
+
+    order = pyblish.api.CollectorOrder
+    label = "Setting up Shotgrid user session."
+
+    def process(self, context):
+        self.log.info("Gathering The Shotgrid Session")
+        certificate_path = os.getenv("SHOTGUN_API_CACERTS")
+
+        if certificate_path is None or not os.path.exists(certificate_path):
+            self.log.info(
+                "SHOTGUN_API_CACERTS does not contains a valid \
+                path: {}".format(
+                    certificate_path
+                )
+            )
+            certificate_path = os.path.join(
+                os.path.dirname(shotgun_api3.__file__),
+                "lib",
+                "certifi",
+                "cacert.pem"
+            )
+            self.log.info("Get Certificate from shotgrid_api")
+
+        if not os.path.exists(certificate_path):
+            self.log.error(
+                "Could not find certificate in shotgun_api3: \
+                {}".format(
+                    certificate_path
+                )
+            )
+            raise ValueError
+
+        self.log.info("Set Shotgrid Certificate: {}".format(certificate_path))
+        os.environ["SHOTGUN_API_CACERTS"] = certificate_path
+
+        shotgrid_module = context.data['openPypeModules']["shotgrid"]
+        user_login = get_local_login()
+        shotgrid_url = shotgrid_module.module_settings.get("shotgrid_server")
+        shotgrid_script_name = shotgrid_module.module_settings.get("shotgrid_script_name")
```

Comment:
Do we know what `shotgrid_script_name` is?

Reply:
> In order to authenticate as a script, your script must be [registered with ShotGrid and have a valid API key](https://developer.shotgridsoftware.com/python-api/authentication.html#setting-up-shotgrid). When creating your shotgun_api3.Shotgun object, provide the script_name and api_key.:

https://developer.shotgridsoftware.com/python-api/authentication.html#script-based-authentication